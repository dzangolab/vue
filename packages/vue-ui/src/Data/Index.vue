<template>
  <div :class="`data direction-${direction}`">
    <div class="data-label">{{ label }}</div>
    <div v-if="slots.separator" class="separator">
      <slot name="separator"></slot>
    </div>
    <div class="data-value">
      <slot name="value">
        {{ displayValue }}
      </slot>
    </div>
  </div>
</template>

<script lang="ts">
export default {
  name: "DataElement",
};
</script>

<script setup lang="ts">
import { computed, useSlots } from "vue";

import type { PropType } from "vue";

const props = defineProps({
  dataKey: {
    default: undefined,
    type: String,
  },
  direction: {
    default: "vertical",
    type: String,
    validator: (value: string) => ["horizontal", "vertical"].includes(value),
  },
  label: {
    type: [String, Number, Object] as PropType<string | number>,
    required: true,
  },
  value: {
    default: "",
    type: [Object, String, Number] as PropType<string | number | object>,
  },
});

const slots = useSlots();

const displayValue = computed(() => {
  if (
    props.dataKey &&
    typeof props.value === "object" &&
    props.value !== null &&
    props.dataKey in props.value
  ) {
    return props.value[props.dataKey as keyof typeof props.value];
  }
  return props.value;
});
</script>

<style lang="css">
@import "../assets/css/data.css";
</style>
