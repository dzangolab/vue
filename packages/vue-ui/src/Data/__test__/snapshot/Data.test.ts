import { mount } from "@vue/test-utils";
import { describe, expect, it } from "vitest";

import Data from "../../Index.vue";

describe("Data", () => {
  it("matches snapshot", () => {
    const wrapper = mount(Data, {
      props: {
        label: "Email",
        value: "john.doe@example.com",
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });
});
