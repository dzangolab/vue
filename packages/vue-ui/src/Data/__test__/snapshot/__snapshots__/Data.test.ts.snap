// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Data > matches snapshot 1`] = `
<div
  class="data direction-vertical"
  label="Email"
>
  <div
    class="label"
  >
    
    
    
  </div>
  <div
    class="value"
  >
    
    john.doe@example.com
    
  </div>
</div>
`;
